mythril/mythril/mythril_leveldb.py

```diff
@@ -0,0 +1,49 @@
+import re
+from mythril.exceptions import CriticalError
+
+
+class MythrilLevelDB:
+    """
+    Class which does search operations on leveldb
+    There are two DBs
+    1) Key value pairs of hashes and it's corresponding address
+    2) The LevelDB Trie
+    """
+
+    def __init__(self, leveldb):
+        """
+
+        :param leveldb: Leveldb path
+        """
+        self.leveldb = leveldb
+
+    def search_db(self, search):
+        """
+        Searches the corresponding code
+        :param search: The code part to be searched
+        """
+
+        def search_callback(_, address, balance):
+            """
+
+            :param _:
+            :param address: The address of the contract with the code in search
+            :param balance: The balance of the corresponding contract
+            """
+            print("Address: " + address + ", balance: " + str(balance))
+
+        try:
+            self.leveldb.search(search, search_callback)
+
+        except SyntaxError:
+            raise CriticalError("Syntax error in search expression.")
```

Comment:
When will this exception be raised? Should we be casting `address` to a string? Can this error message explain why there was a syntax error?

Reply:
it is raised when there is a syntax problem in the leveldb search text, And the error raised is pretty unreliable and useless, like `unexpected EOF while parsing (<string>, line 0)` for all the errors in our case.